x/metadata/keeper/keeper_test.go

```diff
@@ -312,6 +328,107 @@ func (s *KeeperTestSuite) TestValidateAllOwnerPartiesAreSigners() {
+	}{
+		// generic authorization test cases
+		{
+			name: "three parties - one missing signature with one authz grant - two signers",
+			owners: []types.Party{
+				{Address: s.user1, Role: types.PartyType_PARTY_TYPE_OWNER},
+				{Address: s.user2, Role: types.PartyType_PARTY_TYPE_OWNER},
+				{Address: s.user3, Role: types.PartyType_PARTY_TYPE_OWNER}}, // grantee of singer 1
+			signers:               []string{s.user2, s.user3},
+			msgTypeURL:            types.TypeURLMsgWriteScopeRequest,
+			allowedAuthorizations: oneAllowedAuthorizations,
+			granter:               s.user1Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              "",
+		},
+		{
+			name: "three parties - one missing signature with a special case message type - authz grant - two signers",
+			owners: []types.Party{
+				{Address: s.user1, Role: types.PartyType_PARTY_TYPE_OWNER},
+				{Address: s.user2, Role: types.PartyType_PARTY_TYPE_OWNER},
+				{Address: s.user3, Role: types.PartyType_PARTY_TYPE_OWNER}}, // grantee of singer 2
+			signers:               []string{s.user1, s.user3},
+			msgTypeURL:            types.TypeURLMsgAddScopeDataAccessRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               s.user2Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              "",
+		},
+		{
+			name: "three parties - one missing signature with a special case message type - authz grant on parent message type - two signers",
+			owners: []types.Party{
+				{Address: s.user1, Role: types.PartyType_PARTY_TYPE_OWNER},
+				{Address: s.user2, Role: types.PartyType_PARTY_TYPE_OWNER},
+				{Address: s.user3, Role: types.PartyType_PARTY_TYPE_OWNER}}, // grantee of singer 2
+			signers:               []string{s.user1, s.user3},
+			msgTypeURL:            types.TypeURLMsgAddContractSpecToScopeSpecRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               s.user2Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              "",
+		},
+		{
+			name: "two parties - one missing signature with a special case message type without authz grant - one signer",
+			owners: []types.Party{
+				{Address: s.user2, Role: types.PartyType_PARTY_TYPE_OWNER},
+				{Address: s.user3, Role: types.PartyType_PARTY_TYPE_OWNER}},
+			signers:               []string{s.user3},
+			msgTypeURL:            types.TypeURLMsgDeleteRecordRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               nil,
+			grantee:               nil,
+			errorMsg:              fmt.Sprintf("missing signature from [%s (PARTY_TYPE_OWNER)]", s.user2),
+		},
+	}
+
+	now := s.ctx.BlockHeader().Time
+	s.Require().NotNil(now)
+
+	// Test cases
+	for _, tc := range cases {
+		s.T().Run(tc.name, func(t *testing.T) {
+			a := authz.NewCountAuthorization(tc.msgTypeURL, tc.allowedAuthorizations)
+			err := s.app.AuthzKeeper.SaveGrant(s.ctx, tc.grantee, tc.granter, a, now.Add(time.Hour))
+			s.Require().NoError(err)
```

Comment:
Need to skip this if the `granter` or `grantee` is nil. It's also best to include a message in the assertions so that, when it breaks, it's easier to identify. Something like `"SaveGrant"` is all that's needed.

Reply:
Hmm,  just looked at `SaveGrant` and it doesn't check for `nil` `granter` and `grantee`. It doesn't error out either. The goal must be to collect `gas` fees?